orchestrator/src/worker/event_handler/jobs/proof_registration.rs

```diff
@@ -0,0 +1,168 @@
+            format!("Failed to submit proof for L2 verification for job_id: {}, task_id: {}", job.internal_id, task_id),
+        )?;
+
+        tracing::info!(
+            log_type = "completed",
+            category = "proof_registry",
+            function_type = "process_job",
+            job_id = ?job.id,
+            block_no = %internal_id,
+            %external_id,
+            "Proof registration job processed successfully."
+        );
+        Ok(external_id)
+    }
+
+    #[tracing::instrument(fields(category = "proof_registry"), skip(self, config), ret, err)]
+    async fn verify_job(&self, config: Arc<Config>, job: &mut JobItem) -> Result<JobVerificationStatus, JobError> {
+        let internal_id = job.internal_id.clone();
+        tracing::info!(
+            log_type = "starting",
+            category = "proof_registry",
+            function_type = "verify_job",
+            job_id = ?job.id,
+            block_no = %internal_id,
+            "Proof registration job verification started."
+        );
+
+        let task_id: String = job
+            .external_id
+            .unwrap_string()
+            .context(format!("Failed to unwrap external_id for job_id: {}, internal_id: {}", job.id, internal_id))?
+            .into();
+        let proving_metadata: ProvingMetadata = job.metadata.specific.clone().try_into()?;
+        // Determine if we need on-chain verification
+        let (cross_verify, fact) = match &proving_metadata.ensure_on_chain_registration {
+            Some(fact_str) => (true, Some(fact_str.clone())),
+            None => (false, None),
+        };
+
+        tracing::debug!(job_id = %job.internal_id, %task_id, "Getting task status from prover client");
+        let task_status =
+            config.prover_client().get_task_status(&task_id, fact.clone(), cross_verify).await.context(format!(
+                "Failed to get task status from prover client for job_id: {}, task_id: {}",
+                job.internal_id, task_id
+            ))?;
+
+        match task_status {
+            TaskStatus::Processing => {
+                tracing::info!(
+                    log_type = "pending",
+                    category = "proof_registry",
+                    function_type = "verify_job",
+                    job_id = ?job.id,
+                    block_no = %internal_id,
+                    "Proof registration job verification pending."
+                );
+                Ok(JobVerificationStatus::Pending)
+            }
+            TaskStatus::Succeeded => {
+                let fetched_proof =
+                    config.prover_client().get_proof(&task_id, fact.unwrap().as_str()).await.context(format!(
+                        "Failed to fetch proof from prover client for job_id: {}, task_id: {}",
+                        job.internal_id, task_id
+                    ))?;
+
+                let proof_key = format!("{internal_id}/{PROOF_FILE_NAME}");
```

Comment:
we are putting the proof from the 2nd process at the same place as the proof from the 1st process
```suggestion
                let proof_key = format!("{internal_id}/{PROOF_PART2_FILE_NAME}");
```